src/utils/eventListeners.ts

```diff
@@ -0,0 +1,131 @@
+import { ClientType } from '@colony/colony-js';
+import { Log } from '@ethersproject/abstract-provider';
+import { Contract, utils } from 'ethers';
+
+import { addEvent } from '../eventQueue';
+import networkClient from '../networkClient';
+import { ContractEventsSignatures } from '../types';
+import {
+  addEventToQueue,
+  getClientAndProviderForELG as getClientAndProvider,
+  getEventFilter,
+  mapLogToContractEvent,
+} from './events';
+import { verbose } from './logger';
+/**
+ * Generator method for events listeners
+ *
+ * It basically does away with all the boilerplate of setting up topics, setting
+ * the event listener, parsing the received event
+ */
+export const eventListenerGenerator = async (
```

Comment:
Have moved all the listeners to their own file to separate them from the helpers they use, especially now I've written a few for `eventListenerGenerator`